src/lib/entities.ts

```diff
@@ -0,0 +1,123 @@
+import { Authenticator, AuthChain } from 'dcl-crypto'
+import { Personal } from 'web3x-es/personal'
+import { Eth } from 'web3x-es/eth'
+import { Address } from 'web3x-es/address'
+import {
+  CatalystClient,
+  DeploymentBuilder,
+  DeploymentPreparationData
+} from 'dcl-catalyst-client'
+import { Entity, EntityType } from 'dcl-catalyst-commons'
+import { getEthereumProvider } from './eth'
+import { ProfileEntity } from './types'
+import { PeerAPI } from './peer'
+
+export class EntitesOperator {
+  private readonly catalystClient: CatalystClient
+  private readonly peerAPI: PeerAPI
+
+  constructor(peerUrl: string) {
+    this.catalystClient = new CatalystClient(peerUrl, 'builder')
+    this.peerAPI = new PeerAPI(peerUrl)
+  }
+
+  /**
+   * Builds the entity deployment preparation data by preparing
+   * the contents of an entity that doesn't need new files to be
+   * added.
+   *
+   * @param entity - The entity that will be pre-processed prior to its deployment.
+   * @param type - The entity type that will be prepared to be deployed.
+   * @param address - The address of the owner of the entity to be deployed.
+   */
+  private async buildDeployPreparationDataWithoutFiles(
+    entity: Entity,
+    type: EntityType,
+    address: string
+  ): Promise<DeploymentPreparationData> {
+    const content: Map<string, string> = new Map(
+      (entity.content || []).map(({ file, hash }) => [file, hash])
+    )
+
+    return DeploymentBuilder.buildEntityWithoutNewFiles(
+      type,
+      [address],
+      content,
+      entity.metadata
+    )
+  }
+
+  /**
+   * Uses the provider to request the user for a signature to
+   * deploy an entity.
+   *
+   * @param address - The address of the deployer of the entity.
+   * @param entityId - The entity id that it's going to be deployed.
+   */
+  private async authenticateEntityDeployment(
+    address: string,
+    entityId: string
+  ): Promise<AuthChain> {
+    const eth: Eth = await getEthereumProvider()
```

Comment:
We already have a `getNetworkProvider(chainId: ChainId)` to get a provider from any network, or `getConnectedProvider()` to get the provider that the user has connected (disregarding on which network it is connected to).

I would advise against adding a new helper to get a provider, I know this helper returns already a `web3x` eth abstraction, but since we plan to move away from web3x sooner than later, I'd use the `getConnectedProvider` helper and instantiate `Eth` here instead for the signature, like:

```ts
const provider = await getConnectedProvider()
const eth = new Eth(provider!)
```

Reply:
Done!